pkg/controller/invocation/controller.go

```diff
@@ -339,6 +321,37 @@ func (cr *Controller) createFailAction(invocationID string, err error) controlle
 	}
 }
 
+func (cr *Controller) runWorker() {
+	pool := gopool.New(maxParallelExecutions)
+
+	for cr.processNextItem(pool) {
+		// continue looping
+	}
+}
+
+func (cr *Controller) processNextItem(pool *gopool.GoPool) bool {
+	key, quit := cr.workQueue.Get()
+	if quit {
+		return false
+	}
+	defer cr.workQueue.Done(key)
+
+	es := key.(*controller.EvalState)
+
+	err := pool.Submit(cr.Context, func() {
+		controller.EvalQueueSize.WithLabelValues("invocation").Dec()
+		cr.Evaluate(es.ID())
+	})
+	if err != nil {
+		log.Fatalf("failed to submit invocation %v for execution", es.ID())
```

Comment:
This will crash the entire workflows bundle, while the error thrown by the pool.submit is only thrown due to a canceled context. Wouldn't it be better to just log an error here, and continue?